includes/Classifai/Providers/OpenAI/Embeddings.php

```diff
@@ -0,0 +1,722 @@
+		 *
+		 * @param {array} $post_types Array of post statuses to be classified.
+		 *
+		 * @return {array} Array of post statuses.
+		 */
+		return apply_filters( 'classifai_openai_embeddings_post_statuses', $this->get_supported_post_statuses() );
+	}
+
+	/**
+	 * The list of supported taxonomies.
+	 *
+	 * @return array
+	 */
+	public function supported_taxonomies() {
+		/**
+		 * Filter taxonomies supported for embeddings.
+		 *
+		 * @since x.x.x
+		 * @hook classifai_openai_embeddings_taxonomies
+		 *
+		 * @param {array} $taxonomies Array of taxonomies to be classified.
+		 *
+		 * @return {array} Array of taxonomies.
+		 */
+		return apply_filters( 'classifai_openai_embeddings_taxonomies', $this->get_supported_taxonomies() );
+	}
+
+	/**
+	 * Trigger embedding generation for content being saved.
+	 *
+	 * @param int $post_id ID of post being saved.
+	 */
+	public function generate_embeddings_for_post( $post_id ) {
+		// Don't run on autosaves.
+		if ( defined( 'DOING_AUTOSAVE' ) && DOING_AUTOSAVE ) {
+			return;
+		}
+
+		// Ensure the user has permissions to edit.
+		if ( ! current_user_can( 'edit_post', $post_id ) ) {
+			return;
+		}
+
+		$post = get_post( $post_id );
+
+		// Only run on supported post types and statuses.
+		if (
+			! in_array( $post->post_type, $this->supported_post_types(), true ) ||
+			! in_array( $post->post_status, $this->supported_post_statuses(), true )
+		) {
+			return;
+		}
+
+		// Don't run if turned off for this particular post.
+		if ( 'no' === get_post_meta( $post_id, '_classifai_process_content', true ) ) {
+			return;
+		}
+
+		$embeddings = $this->generate_embeddings( $post_id, 'post' );
+
+		// Add terms to this item based on embedding data.
+		if ( $embeddings && ! is_wp_error( $embeddings ) ) {
+			update_post_meta( $post_id, 'classifai_openai_embeddings', array_map( 'sanitize_text_field', $embeddings ) );
+			$this->set_terms( $post_id, $embeddings );
+		}
+	}
+
+	/**
+	 * Add terms to a post based on embeddings.
+	 *
+	 * @param int   $post_id ID of post to set terms on.
+	 * @param array $embedding Embedding data.
+	 */
+	private function set_terms( int $post_id = 0, array $embedding = [] ) {
+		if ( ! $post_id || ! get_post( $post_id ) ) {
+			return new WP_Error( 'post_id_required', esc_html__( 'A valid post ID is required to set terms.', 'classifai' ) );
+		}
+
+		if ( empty( $embedding ) ) {
+			return new WP_Error( 'data_required', esc_html__( 'Valid embedding data is required to set terms.', 'classifai' ) );
+		}
+
+		$settings             = $this->get_settings();
+		$number_to_add        = $settings['number'] ?? 1;
+		$embedding_similarity = [];
+		$taxonomies           = $this->supported_taxonomies();
+		$calculations         = new EmbeddingCalculations();
+
+		foreach ( $taxonomies as $tax ) {
+			$terms = get_terms(
+				[
+					'taxonomy'   => $tax,
+					'hide_empty' => false,
+					'fields'     => 'ids',
+					'meta_key'   => 'classifai_openai_embeddings',
```

Comment:
So it is intentional here because this method is comparing existing term data to post data to determine which terms to set; we're not generating embedding data for terms.

That said, looking at the other places in this file that we get term information, I did unintentionally set the meta key in the `trigger_taxonomy_update` method, which maybe you meant to comment on that line instead of here. That is now fixed so we generate embedding data for any terms that don't have that meta key set yet.

Reply:
Yes, it was meant for meta_key args in the `trigger_taxonomy_update` function. Thanks for fixing it.